routers/web/repo/issue_suggestions.go

```diff
@@ -0,0 +1,82 @@
+// Copyright 2024 The Gitea Authors. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+package repo
+
+import (
+	"net/http"
+	"strings"
+
+	"code.gitea.io/gitea/models/db"
+	issues_model "code.gitea.io/gitea/models/issues"
+	issue_indexer "code.gitea.io/gitea/modules/indexer/issues"
+	"code.gitea.io/gitea/services/context"
+)
+
+type issueSuggestion struct {
+	ID          int64  `json:"id"`
+	Title       string `json:"title"`
+	State       string `json:"state"`
+	PullRequest *struct {
+		Merged bool `json:"merged"`
+		Draft  bool `json:"draft"`
+	} `json:"pull_request,omitempty"`
+}
+
+// IssueSuggestions returns a list of issue suggestions
+func IssueSuggestions(ctx *context.Context) {
+	keyword := strings.ToLower(ctx.Req.FormValue("q"))
```

Comment:
By reading the code, I think it shouldn't do `ToLower`

Because there is a DB indexer, while the content in DB is case-sensitive.

If you do `ToLower` here, then the contents with upper-case would never be correctly fetched.

So, the best approach is to do nothing, do not `ToLower`, and leave the details to the indexers.